numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -95,6 +95,11 @@ def __init__(
                 capacity=config.CACHE_SIZE,
                 pyfunc=self.pyfunc,
             )
+            self._kernel_bundle_cache = LRUCache(
+                name="KernelBundleCache",
+                capacity=config.CACHE_SIZE,
+                pyfunc=self.pyfunc,
+            )
         else:
             self._cache = NullCache()
         self._cache_hits = 0
```

Comment:
Is _cache_hits being used? if this is only for book-keeping during debugging then it needs to be removed.

Reply:
Yea it's only for debugging, but we use it in unit tests though.

Reply:
Can we then just use it in debug mode by wrapping it in "if config.DEBUG_CACHE:"? Although a single update overhead can add up.

Reply:
I doubt updating the flag is having any discernible overhead.